apps/mobile/src/features/ConfirmTx/components/SignTransaction/SignTransaction.tsx

```diff
@@ -6,27 +6,44 @@ import { selectChainById } from '@/src/store/chains'
 import { RootState } from '@/src/store'
 import type { ChainInfo } from '@safe-global/safe-gateway-typescript-sdk'
 import { useLocalSearchParams } from 'expo-router'
-import { useLazySignTransactionQuery } from '@/src/store/signersBalance'
 import { getPrivateKey } from '@/src/hooks/useSign/useSign'
 import SignError from './SignError'
 import SignSuccess from './SignSuccess'
+import { signTx } from '@/src/services/tx/tx-sender/sign'
+import { useTransactionsAddConfirmationV1Mutation } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
+import logger from '@/src/utils/logger'
 
 export function SignTransaction() {
   const { txId, signerAddress } = useLocalSearchParams<{ txId: string; signerAddress: string }>()
   const activeSafe = useDefinedActiveSafe()
   const activeChain = useAppSelector((state: RootState) => selectChainById(state, activeSafe.chainId))
 
-  const [trigger, { isFetching, data, isError }] = useLazySignTransactionQuery()
+  const [addConfirmation, { isLoading, data, isError }] = useTransactionsAddConfirmationV1Mutation()
 
   const sign = useCallback(async () => {
-    const privateKey = await getPrivateKey(signerAddress)
-
-    trigger({
-      chain: activeChain as ChainInfo,
-      activeSafe,
-      txId,
-      privateKey,
-    })
+    try {
+      const privateKey = await getPrivateKey(signerAddress)
+
+      const signedTx = await signTx({
+        chain: activeChain as ChainInfo,
```

Comment:
Nit: can this be achieved with a type guard instead?

Reply:
we're having some types conflict in the app 😞 We will address it in a separated PR soon